dl1_data_handler/utils.py

```diff
@@ -0,0 +1,98 @@
+import numpy as np
+import ctapipe.image.cleaning as cleaning
+from ctapipe.instrument import CameraGeometry
+
+
+#################
+# Event Filters #
+#################
+
+def event_energy_filter(reader, file, e_min=-np.inf, e_max=np.inf):
+    """
+    Filter events on energy
+    Parameters
+    ----------
+    reader
+    file
+    e_min
+    e_max
+
+    Returns
+    -------
+    the filtered indices
+    """
+    mask1 = (file.root.Events[:]['mc_energy'] > e_min) & (file.root.Events[:]['mc_energy'] < e_max)
+    return set(np.arange(len(file.root.Events))[mask1])
+
+
+def event_intensity_filter(reader, file, i_min=-np.inf, i_max=np.inf):
+    """
+    Filter events on intensity (in pe)
+    Parameters
+    ----------
+    reader (DL1DataReader) : the reader to filter
+    file
+    i_min
+    i_max
+
+    Returns
+    -------
+    the filtered indices
+    """
+    tel_types = [reader.tel_type] if reader.mode == 'mono' else reader.tel_type
+    total_intensity = np.zeros(len(file.root.Events))
+    for tel_type in tel_types:
+        indices = file.root.Events[:][tel_type + '_indices']
+        images = file.root[tel_type][:]['charge']
```

Comment:
```suggestion
        images = file.root._f_get_child(tel_type)[:]['charge']
```

Reply:
What is the difference between calling `._f_get_child()` and `[name_of_child]` ?

Reply:
The second syntax works for getting columns from a table programmatically, but not branches from the file.

Reply:
ok thank you